examples/plugins/README.md

```diff
@@ -69,16 +86,131 @@ For example:
-* `interfaces` is a list of interfaces which this plugin supports. It is required, and must equal `["reporter"]`.
+* `description` is displayed in the UI.
+* `interfaces` is a list of interfaces which this plugin supports. It is required, and must contain at least `["reporter"]`.
 * `api_version` is used to ensure both the plugin and the scope probe can speak to each other. It is required, and must match the probe.
 
-### <a id="interfaces"></a>Interfaces
+You may notice a small chicken and egg problem - the plugin reports to
+the scope probe what interfaces it supports, but the scope probe can
+learn that only by doing a `GET /report` request which will be handled
+by the plugin if it implements the "reporter" interface. This is
+solved (or worked around) by requiring the plugin to always implements the
+"reporter" interface.
+
+### Controller interface
+
+Implementing the controller interface means that the plugin can react
+to HTTP `POST` control requests sent by the app. The plugin will
+receive them only for controls it exposed in its reports. The requests
+will come to the `/control` endpoint.
+
+Add the "controller" string to the `interfaces` field in the plugin
+specification.
+
+#### Control
+
+The `POST` requests will have a JSON-encoded body with the following contents:
+
+```json
+{
+  "AppID": "some ID of an app",
+  "NodeID": "an ID of the node that had the control activated",
+  "Control": "the name of the activated control"
+}
+```
+
+The body of the response should also be a JSON-encoded data. Usually
+the body would be an empty JSON object (so, "{}" after
+serialization). If some error happens during handling the control,
+then the plugin can send a response with an `error` field set, for
+example:
+
+```json
+{
+  "error": "An error message here"
+}
+```
+
+Sometimes the control activation can make the control obsolete, so the
+plugin may want to hide it (for example, control for stopping the
+container should be hidden after the container is stopped). For this
+to work, the plugin can send a shortcut report by filling the
+`ShortcutReport` field in the response, like for example:
+
+```json
+{
+  "ShortcutReport": { body of the report here }
+}
+```
+
+##### How to expose controls
+
+Each topology in the report (be it host, pod, endpoint and so on) has
+a set of available controls a node in the topology may want to
+show. The following (rather artificial) example shows a topology with
+two controls (`ctrl-one` and `ctrl-two`) and two nodes, each having a
+different control from the two:
+
+```json
+{
+  "Host": {
+    "controls": {
+      "ctrl-one": {
+        "id": "ctrl-one",
+        "human": "Ctrl One",
+        "icon": "fa-futbol-o",
+        "rank": 1
+      },
+      "ctrl-two": {
+        "id": "ctrl-two",
+        "human": "Ctrl Two",
+        "icon": "fa-beer",
+        "rank": 2
+      }
+    },
+    "nodes": {
+      "host1": {
+        "latestControls": {
+          "latest": {
```

Comment:
Hmm, yeah. `latestControls` is a separate struct that has the `report.LatestMap` member. I did it that way in case we want to add something more to `latestControls`… Would you prefer to merge them or a better name would suffice?